oqs/src/macros.rs

```diff
@@ -31,7 +31,7 @@ macro_rules! newtype_buffer {
 
         impl<'a> $name_ref<'a> {
             /// Construct a new container around this reference version
-            fn new(bytes: &'a [u8]) -> $name_ref<'a> {
+            pub fn new(bytes: &'a [u8]) -> $name_ref<'a> {
```

Comment:
Implementing this requires an assertion on the length of `bytes`.

Reply:
True, especially if we're dealing with pointer arithmetic under the hood. In idiomatic rust would use result types instead of assertions, especially if we decide to hand over partial control of building these wrappers to the user.